src/main/java/PieceCopier.java

```diff
@@ -0,0 +1,33 @@
+public class PieceCopier {
+    /**
+     * Returns a deep clone of the given Piece instance.
+     *
+     * @see Piece#copy() Piece.copy()
+     *
+     * @param piece the board to be cloned
+     * @return a clone of the board
+     */
+
+    private Piece newPiece = new Piece();
+    private Piece oldPiece;
+
+    public PieceCopier(Piece oldPiece) {
```

Comment:
Is it necessary to lock every PieceCopier to the piece it's copying? I can't imagine that a PieceCopier is ever used more than once in this paradigm. It could be better to allow PieceCopier to have just one method `Piece copy(Piece piece) {...}` so that a single PieceCopier instance can be used to copy all the pieces.